Add `cvmfs_apt_repository_url` to enable the use of a mirror repository

jsl6ul opened the pull request.

jsl6ul committed:
Add `cvmfs_apt_repository_url` to enable the use of a mirror repository

btravouillon commented on defaults/main.yml:
```diff
@@ -1,6 +1,8 @@
 ---
 # defaults file for galaxyproject.cvmfs
 
+cvmfs_apt_repository_url: https://cvmrepo.s3.cern.ch/cvmrepo/apt/
```
Could you use `cvmfs_apt_repository` instead. This is the format we use in most of our other roles.
Also, it may be better to define the whole deb line so we could tweak it further if required.
```suggestion
cvmfs_apt_repository: "deb [allow-insecure=true] https://cvmrepo.s3.cern.ch/cvmrepo/apt/ {{ ansible_distribution_release }}-prod main"
```

jsl6ul replied:
Defining the whole deb line won't work with the 3rd task at line 32.
`repo: deb [allow-insecure=true] {{ cvmfs_apt_repository_url }} xenial-prod main`

jsl6ul replied:
or I can change the repo for the first 2 tasks, and ignore the 3rd one and keep it hard-coded to cvmrepo.s3.cern.ch...?

btravouillon replied:
This is legacy code for Ubuntu releases older than Xenial (16.04). It could have been useful with static URLs, but it seems irrelevant now. I guess you could simply remove this task from the role.

I also wonder if the task at line 19 is still relevant. I guess you could remove the `when: ansible_distribution != 'Ubuntu'` in the first task and let the user define something different from the default if required.

Am I missing something?

jsl6ul committed:
Change cvmfs_apt_repository_url to cvmfs_apt_repository

jsl6ul committed:
Delete duplicated tasks: ansible.builtin.apt_repository

Repository exceptions can be managed in host_vars and/or group_vars
using `cvmfs_apt_repository`

jsl6ul committed:
Fix repository file mode

btravouillon reviewed (approved):
Looks good to me. 👍
